solidity/contracts/token/HypNative.sol

```diff
@@ -17,90 +32,120 @@ contract HypNative is TokenRouter {
-        require(msg.value >= _amount, "Native: amount exceeds msg.value");
-        uint256 _hookPayment = msg.value - _amount;
-        return _transferRemote(_destination, _recipient, _amount, _hookPayment);
-    }
+    // ============ External Functions ============
 
     /**
      * @inheritdoc TokenRouter
-     * @dev uses (`msg.value` - `_amount`) as hook payment.
+     * @dev use _hook with caution, make sure that this hook can handle msg.value transfer using the metadata.msgValue()
      */
     function transferRemote(
         uint32 _destination,
         bytes32 _recipient,
         uint256 _amount,
         bytes calldata _hookMetadata,
         address _hook
-    ) external payable virtual override returns (bytes32 messageId) {
-        require(msg.value >= _amount, "Native: amount exceeds msg.value");
-        uint256 _hookPayment = msg.value - _amount;
+    ) public payable virtual override returns (bytes32 messageId) {
+        uint256 quote = _GasRouter_quoteDispatch(
+            _destination,
+            _hookMetadata,
+            _hook
+        );
+        require(msg.value >= _amount + quote, "HypNative: insufficient value");
+
+        bytes memory hookMetadata = StandardHookMetadata.overrideMsgValue(
+            _hookMetadata,
+            _amount
+        );
+
         return
             _transferRemote(
                 _destination,
                 _recipient,
                 _amount,
-                _hookPayment,
-                _hookMetadata,
+                _amount + quote,
+                hookMetadata,
                 _hook
             );
     }
 
-    function balanceOf(
-        address _account
-    ) external view override returns (uint256) {
-        return _account.balance;
+    /// @inheritdoc TokenRouter
+    function transferRemote(
+        uint32 _destination,
+        bytes32 _recipient,
+        uint256 _amount
+    ) external payable virtual override returns (bytes32 messageId) {
+        bytes calldata emptyBytes;
+        assembly {
+            emptyBytes.length := 0
+            emptyBytes.offset := 0
+        }
+        return
+            transferRemote(
+                _destination,
+                _recipient,
+                _amount,
+                emptyBytes,
+                address(hook)
+            );
     }
 
+    // ============ Internal Functions ============
+
     /**
      * @inheritdoc TokenRouter
-     * @dev No-op because native amount is transferred in `msg.value`
-     * @dev Compiler will not include this in the bytecode.
+     * @dev No token metadata is needed for value transfers
      */
     function _transferFromSender(
         uint256
     ) internal pure override returns (bytes memory) {
-        return bytes(""); // no metadata
+        return bytes(""); // no token metadata
     }
 
     /**
-     * @dev Sends `_amount` of native token to `_recipient` balance.
      * @inheritdoc TokenRouter
+     * @dev Sends the value to the recipient
```

Comment:
why change this?

Reply:
I didn't, this is a new addition. I think you're confusing between the HypNative and HypNativeCollateral.